llvm/include/llvm/IR/InstrTypes.h

```diff
@@ -912,6 +912,16 @@ class CmpInst : public Instruction {
   /// Determine if this is an equals/not equals predicate.
   bool isEquality() const { return isEquality(getPredicate()); }
 
+  /// Determine if this is an equals predicate that is also an equivalence. This
+  /// is useful in GVN-like transformations, where we can replace RHS by LHS in
+  /// the true branch of the CmpInst.
```

Comment:
Rather than "this is useful", this should explain what "equivalence" here is supposed to mean. You'll likely have to add a caveat that this ignores provenance considerations.

Reply:
I'm not sure what the caveat is, if the operands are loaded from memory: can you explain?

Reply:
For pointers, if A == B, you generally cannot replace A with B unless they have the same provenance.